include/ensmallen_bits/sdp/lin_alg.hpp

```diff
@@ -15,6 +15,38 @@
 namespace ens {
 namespace math {
 
+/**
+* Matrix representation of a general Linear operator representated 
+* by a function/method.
+*
+* @param input Input function.
+* @param output The matrix representation.
+*/
+template<typename ElemType, typename MatBType>
+inline void convertToMatrix(std::function<ElemType(arma::Mat<ElemType>)> input, 
+                            MatBType& output)
```

Comment:
It looks like the approach you are taking is to turn a `std::function` that the user gives into a matrix, and then use that as a constraint.  But this is the opposite of what we are looking for in #72: take the given example of the constraint that the sum of elements is a constant.  You can express this very tediously as the constraint `trace(1 1^T X) = b`, but the much easier way to represent this is `sum(X) = b`.  So, instead of extracting the matrix `1 1^T`, it would be better to just store the `std::function`s in the SDP objects and evaluate them directly to get the value.

Reply:
First of all, thank you for taking the time to look into this. I'm glad you asked this. I honestly didn't know if this would be alright.

However, (from what I understand) in the interior point method, the constraints need to be expressed as matrices (as matrix multiplication and transposing is involved).

Besides this, in the LRSDP implementation, I have used these `std::function`s wherever the matrix conversion isn't necessary (again like in the case of matrix multiplication or transposing).